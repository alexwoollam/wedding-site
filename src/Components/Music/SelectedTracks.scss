@import '../../variables';

.btn-remove{
  background-color: $dustypink;
  border: 2px solid $dustypink--dark;
  &:hover{
    background-color: $dustypink--dark;
  }
}
