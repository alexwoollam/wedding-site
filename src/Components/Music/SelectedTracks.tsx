import React from 'react';
import {Container, Row, Col, Button} from 'reactstrap';

interface PropsDefined {
    data: any,
    addTrack?: any,
    removeTrack?: any,
    isSelected?: boolean,
}

function SelectedTracks( props: PropsDefined ){

    if( ! props.data ){ 
        return null;
    }

    let items = props.data;

    if( items.length > 0 ){
        return (
            <Container className={ 'mb-3' }>
                {items.map( (track, index)=>(
                    <Row key={ index } style={{border: '2px dashed lightgrey'}} className='p-3 mt-2 rounded'>
                        <Col md={1}>
                            <img className={'rounded'} src={ track.trackimage } alt={"artwork for " + track.trackname } />
                        </Col>
                        <Col>
                            <h4>{ track.trackname }</h4>
                            <h5>{ track.trackartist }</h5>
                        </Col>
                        <Col md={1} className="d-flex justify-content-center">
                            <Button data-testid='remove-button' className={'btn-danger'} onClick={ () => props.removeTrack(track.id) }>Remove</Button>
                        </Col>
                    </Row>
                ))}
            </Container>
        );
    } else {
        return null;
    }
} 

export default SelectedTracks;