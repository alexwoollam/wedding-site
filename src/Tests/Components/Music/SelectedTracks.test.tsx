//ts-noscan
import '@testing-library/jest-dom';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import SelectedTracks from '../../../Components/Music/SelectedTracks';


// @ts-ignore
Enzyme.configure({ adapter: new Adapter() });

describe('<SelectedTrack/> component', () => {

    let MockTracks = {};
    MockTracks = [
            {
                'id': 0,
                'trackid': 'abc',
                'trackname': 'Song 2',
                'trackartist': 'Blur',
                'trackimage': 'image.png',
            },
        ];

        let MockTracksEmpty = {};
        MockTracksEmpty = [];


    test('Component should reuturn null if no props passed', () => {
        const comp = shallow(<SelectedTracks  data={ MockTracksEmpty }/>);
        expect(comp.type()).toEqual(null)
    });

    test('Component should reuturn null if props is empty', () => {
        const comp = shallow(<SelectedTracks data={ MockTracksEmpty } />);
        expect(comp.type()).toEqual(null)
    });

    test('Component should reuturn the correct artist', () => {
        render(<SelectedTracks data={ MockTracks } />);
        const linkElement = screen.getByText('Blur');
        expect(linkElement).toBeInTheDocument();
    });

    test('Component should reuturn the correct track title', () => {
        render(<SelectedTracks data={ MockTracks } />);
        const linkElement = screen.getByText('Song 2');
        expect(linkElement).toBeInTheDocument();
    });

    test('Component should have a remove button', () => {
        render(<SelectedTracks data={ MockTracks } />);
        const linkElement = screen.getByTestId('remove-button');
        expect(linkElement).toBeInTheDocument();
    });

    test('If we remove an element, it should call the removeTrack props', () => {
        const mockRemove = jest.fn();
        const comp = shallow(<SelectedTracks data={ MockTracks } removeTrack={mockRemove} />);
        comp.find('[data-testid="remove-button"]').simulate('click');
        expect(mockRemove.mock.calls.length).toBe(1);
    });

    it('renders correctly', () => {
        const tree = renderer
        .create(<SelectedTracks data={ MockTracks } />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
