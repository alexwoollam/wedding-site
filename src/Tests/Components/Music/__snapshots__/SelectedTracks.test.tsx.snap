// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectedTrack/> component renders correctly 1`] = `
<div
  className="mb-3 container"
>
  <div
    className="p-3 mt-2 rounded row"
    style={
      Object {
        "border": "2px dashed lightgrey",
      }
    }
  >
    <div
      className="col-md-1"
    >
      <img
        alt="artwork for Song 2"
        className="rounded"
        src="image.png"
      />
    </div>
    <div
      className="col"
    >
      <h4>
        Song 2
      </h4>
      <h5>
        Blur
      </h5>
    </div>
    <div
      className="d-flex justify-content-center col-md-1"
    >
      <button
        aria-label={null}
        className="btn-remove btn btn-secondary"
        data-testid="remove-button"
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
